### PR TITLE
New redis fun

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: cauldron/redis_cache.py [hunk 1]
```diff
@@ -1,21 +1,57 @@
 import asyncio
 import aioredis
+from asyncio import coroutine
```
Comment (nerandell) [inline cauldron/redis_cache.py:3]: I would prefer arranging them alphabetically.

File: cauldron/redis_cache.py [hunk 2]
```diff
@@ -1,21 +1,57 @@
...
+    def get_pool(cls):
+        if not cls._pool:
+            with (yield from cls._lock):
+                if not cls._pool:
+                    cls._pool = yield from aioredis.create_pool((cls._host, cls._port), minsize=cls._minsize,
+                                                                maxsize=cls._maxsize)
+        return (yield from cls._pool)
+
+    @classmethod
+    def connect_v2(cls, host, port, minsize=5, maxsize=10, loop=None):
```
Comment (nerandell) [inline cauldron/redis_cache.py:29]: What value does this method add here over the old method?

**Comment by ghost** [comment]
It was created by mistake.

File: cauldron/sql.py [hunk 3]
```diff
@@ -125,10 +130,11 @@ class PostgresStore:
     _WHERE_AND = '{} {} %s'
     _PLACEHOLDER = ' %s,'
     _COMMA = ', '
+    _pool_pending = asyncio.Semaphore(1)
 
     @classmethod
     def connect(cls, database: str, user: str, password: str, host: str, port: int, *, use_pool: bool=True,
-                enable_ssl: bool=False, minsize=1, maxsize=50, keepalives_idle=5, keepalives_interval=4, echo=False,
+                enable_ssl: bool=False, minsize=1, maxsize=30, keepalives_idle=5, keepalives_interval=4, echo=False,
```
Comment (nerandell) [inline cauldron/sql.py:137]: Not sure why this change is essential. The consumer can override the default behaviour.

File: cauldron/sql.py [hunk 4]
```diff
@@ -332,19 +340,32 @@ def select(cls, cur, table: str, order_by: str, columns: list=None, where_keys:
             columns_string = cls._COMMA.join(columns)
             if where_keys:
                 where_clause, values = cls._get_where_clause_with_values(where_keys)
-                query = cls._select_selective_column_with_condition.format(columns_string, table, where_clause,
-                                                                           order_by, limit, offset)
+                if order_by:
+                    query = cls._select_selective_column_with_condition_and_order_by.format(columns_string, table, where_clause,
```
Comment (nerandell) [inline cauldron/sql.py:344]: There is too much duplication that I see here (`if order_by:` condition everywhere). Best approach would be write a DSL to generate MySQL queries rather than keeping hardcoded strings. However this is definitely beyond the scope of this PR. I will keep that as something to fix in future.

**Comment by nerandell** [comment]
Not a problem @prashant70, I hope vyked is running fine on production 😉

**Comment by ghost** [comment]
Thanks Ankit. Yes, vyked is running fine :-)

Thanks,
Prashant Mishra
